backend/functions/src/emailing/functions.ts

```diff
@@ -119,9 +166,20 @@ export const sendMails = async (
   message: any,
   authToken: string,
   courseId: string,
-  group: string,
-  template = 'Share matched results'
+  group = '-1',
+  template = 'Share matched results',
+  indivEmail?: string
 ) => {
+  if (parseInt(group) < 0 && indivEmail === undefined) {
+    logger.error(
+      ` Invalid group ${group} and invalid email "${indivEmail}" for updating timestamps `
+    )
+    return
+  } else if (parseInt(group) > 0 && indivEmail !== undefined) {
+    logger.error(`group and individual email cannot both be specified `)
+    return
+  }
```

Comment:
Specific to the validation logic, I think throwing errors would be better instead of returning void. It plays better with promises (the `.catch` on the promise will execute) and makes the return type of this function better defined, because now it might return void or a number, and it's not explicit that void is an expected return type or what it means (compared to an error).

Also may want to make the logic stronger to check that only one of group or indivEmail can be defined and are valid. Maybe make the group optional instead of default to -1, or something similar idk. Just something to better mirror the request JSON because now mentally we have to take into account that indivEmail will be the same as the request (a string or undefined) but the group number will always exist in this function, which may be the same as the request or maybe not (since it's default to -1).